src/lib.rs

```diff
@@ -483,36 +437,43 @@ impl Unicorn {
         }
     }
 
-    /// Write an unsigned value register.
-    ///
-    /// Note : The register is defined as an i32 to be able to support the
-    /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
-    /// You need to cast the register with `as i32`.
-    pub fn reg_write(&self, regid: i32, value: u64) -> Result<(), Error> {
-        let p_value: *const u64 = &value;
-        let err = unsafe { uc_reg_write(self.handle, regid, p_value as *const libc::c_void) };
+    unsafe fn reg_write_generic<T: Sized>(&self, regid: i32, value: T) -> Result<()> {
+        let p_value: *const T = &value;
+        let err = uc_reg_write(self.handle, regid, p_value as *const libc::c_void);
         if err == Error::OK {
             Ok(())
         } else {
             Err(err)
         }
     }
+
+    /// Write an unsigned value register.
+    ///
+    /// Note : The register is defined as an i32 to be able to support the
+    /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
+    /// You need to cast the register with `as i32`.
+    pub fn reg_write(&self, regid: i32, value: u64) -> Result<()> {
+        unsafe { Self::reg_write_generic::<_>(&self, regid, value) }
+    }
+
     /// Write a signed 32-bit value to a register.
     ///
     /// Note : The register is defined as an i32 to be able to support the
     /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
     /// You need to cast the register with `as i32`.
-    pub fn reg_write_i32(&self, regid: i32, value: i32) -> Result<(), Error> {
-        let p_value: *const i32 = &value;
-        let err = unsafe {
-            uc_reg_write(
-                self.handle,
-                regid as libc::c_int,
-                p_value as *const libc::c_void,
-            )
-        };
+    pub fn reg_write_i32(&self, regid: i32, value: i32) -> Result<()> {
+        unsafe { Self::reg_write_generic::<_>(&self, regid, value) }
+    }
+
+    unsafe fn reg_read_generic<T: Sized + Copy>(&self, regid: i32) -> Result<T> {
+        let mut value: T = mem::zeroed();
```

Comment:
This should work for now, but please add a comment to use `MaybeUninit` when stable.

Reply:
Sorry, sorry (still don't understand why I forgot your comment). 
I've changed in the force update.